packages/hardhat-utils/src/bigint.ts

```diff
@@ -0,0 +1,88 @@
+import { BigIntError } from "./errors/bigint.js";
+import { isBNBigInt, isBigNumberBigInt } from "./internal/bigint.js";
+import { unreachable } from "./errors/catch-utils.js";
+
+/**
+ * Returns the minimum of two BigInts.
+ *
+ * @param x The first number to compare.
+ * @param y The second number to compare.
+ * @returns The smaller of the two numbers.
+ */
+export function min(x: bigint, y: bigint): bigint {
+  return x < y ? x : y;
+}
+
+/**
+ * Returns the maximum of two BigInts.
+ *
+ * @param x The first number to compare.
+ * @param y The second number to compare.
+ * @returns The larger of the two numbers.
+ */
+export function max(x: bigint, y: bigint): bigint {
+  return x > y ? x : y;
+}
+
+/**
+ * Converts a value to a BigInt.
+ *
+ * This function supports several types of input:
+ * - `number`: Must be an integer and a safe integer. If it's not, an error is thrown.
+ * - `bigint`: Returned as is.
+ * - `string`: Converted to a BigInt using the BigInt constructor.
+ * - `object`: Must be an instance of BN or BigNumber. If it's not, an error is thrown.
+ *
+ * If the input is of an unsupported type, an error is thrown.
+ *
+ * @param x The value to convert to a BigInt.
+ * @returns The input value converted to a BigInt.
+ * @throws BigIntError If the input value cannot be converted to a BigInt.
+ */
+export async function toBigInt(
+  x: number | string | bigint | object,
+): Promise<bigint> {
+  switch (typeof x) {
+    case "number":
+      if (!Number.isInteger(x)) {
+        throw new BigIntError(`${x} is not an integer`);
+      }
+      if (!Number.isSafeInteger(x)) {
```

Comment:
This doesn't make sense, as the way to handle unsafe integer is casting them to BigInt, which is what this function does.